Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+        for df, initial_role in zip((X_df, Y_df, meta_df), ('x', 'y', 'meta')):
+            for column_name, column in ((c, df[c]) for c in df.columns):
+                t, r = Domain.infer_type_role(column, initial_role if not infer_roles else None)
+                name = Domain.infer_name(t, r, column_name,
+                                         role_vars['x'], role_vars['y'], role_vars['meta'])
+                if t is DiscreteVariable:
+                    var = t(name, values=DiscreteVariable.generate_unique_values(column))
+                else:
+                    var = t(name)
+                result[var.name] = column
+                role_vars[r].append(var)
+        result.domain = Domain(role_vars['x'], role_vars['y'], role_vars['meta'])
+        result.domain.anonymous = True
+        result._transform_discrete_into_categorical()
+        result._transform_timevariable_into_datetime()
+        return result
+
+    @classmethod
+    def from_numpy(cls, domain, X, Y=None, metas=None, weights=None):
+        """Construct a table from numpy arrays with the given domain.
+
+        The number of variables in the domain must match the number of columns in the
+        corresponding arrays. All arrays must have the same number of rows.
+
+        Parameters
+        ----------
+        domain : Domain | None
+            If None, the domain is inferred from the data. Otherwise, specifies
+            the column assignment to the new Table.
+        X : np.ndarray
+            The X component of the data (or undetermined, depending on the domain).
+        Y : np.ndarray, optional, default None
+            The Y component of the data.
+        metas : np.ndarray, optional, default None
+            The meta attributes of the data.
+        weights : anything TableBase.set_weights accepts, optional, default None
+            The weights to use for the resulting Table.
+        Returns
+        -------
+        TableBase
+            A new Table constructed from the given data.
+        """
+        def correct_shape(what):
+            if what is None or len(what.shape) == 2:
+                return what
+            else:
+                return np.atleast_2d(what).T
+
+        # if anything is sparse, use the sparse version
+        from .impl import SparseTable
+        if sp.issparse(X) or (Y is not None and sp.issparse(Y)) \
+                          or (metas is not None and sp.issparse(metas)) \
+                          or (weights is not None and sp.issparse(weights))\
+                          or issubclass(cls, SparseTable):
+            # explicitly construct a sparse table as this can be called from Table(...),
+            # but allow subclasses to call correctly
+            if issubclass(cls, SparseTable):
+                return cls._from_sparse_numpy(domain, X, Y, metas, weights)
+            else:
+                return SparseTable._from_sparse_numpy(domain, X, Y, metas, weights)
```

Comment:
Not sure what you mean. Corpus is already sparse, this if just exists to allow Table(..) to construct `SparseTable` magically, while still allowing subclasses of `SparseTable` to retain their type.